src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -12,7 +14,13 @@ interface Props {
 
 function HeatmapVis(props: Props): JSX.Element {
   const { dims, data } = props;
-  const textureData = useMemo(() => computeTextureData(data), [data]);
+  const values = data.flat();
+  const interpolator = interpolateMagma;
+  const scale = generateColorScale(values, interpolator);
+  const textureData = useMemo(() => computeTextureData(values, scale), [
+    scale,
+    values,
+  ]);
```

Comment:
The computation of the `scale` was moved in a dedicated function as it is needed by the color bar and was therefore removed from `useMemo`. It is apparently working fine but we could put it back inside.

Reply:
I think you can even remove `useMemo` altogether. I was wondering about it, but since we already use `React.memo` on `HeatmapVis` itself, `computeTextureData` will always be called with new values.